### ISSUE TITLE
Determining parameters for GET/POST/PUT/DELETE methods

### ISSUE THREAD

**Issue by AndrewSouthpaw** [SaltwaterC/http-request#27]
In the documentation, the signature for each is something like `get(options, file, callback)`. In the examples, however, it doesn't always follow this pattern.

```
http.get('http://localhost/get', function (err, res) {
    if (err) {
        console.error(err);
        return;
    }   
    console.log(res.code, res.headers, res.buffer.toString());
});
```

In this case, it follows the pattern `get(file, callback)`. Is this intentional? Is the `options` parameter optional and your code intuits when there is not `options` passed in?

**Comment by SaltwaterC** [comment]
TL;DR it's a feature introduced with [v0.5](https://github.com/SaltwaterC/http-request/blob/master/CHANGELOG.md#v05). The idea isn't new as the node.js http library [does the same since 0.10](http://nodejs.org/api/http.html#http_http_request_options_callback). However, due to legacy reasons, it wasn't properly documented.

**The long version**

That's the idea that sparked http-get (which became http-request), the fact that I needed  to wrap url.parse() with http.get() and make them speak the same keys in order to pass a String containing an URL, hence a wrapper function appeared in my code for this purpose. The next step was adding file downloads and gzip support, hence a new module appeared. The syntax wasn't flexible, hence I opted for an options argument that could be extended. It slipped my mind the idea of passing a String or Object, until node.js did it. Internally, a String is converted to `options = {url: options}`.

The documentation didn't include the shorthand syntax for legacy reasons, although I mentioned it as a comment in the examples. At the time it was implemented, there was a migration from String to Buffer as the default (and only) supported buffered response. Using shorthand without warnings wasn't possible for buffered responses, as shown in #22. It was possible to use the new syntax only for file downloads.

Now that the migration is fully complete, I can properly document the changes, but I simply forgot abut it.

**Comment by AndrewSouthpaw** [comment]
Thanks for the explanation!
